app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala

```diff
@@ -19,50 +19,70 @@ package uk.gov.hmrc.securemessage.services
 import cats.data.NonEmptyList
 import com.google.inject.Inject
 import org.joda.time.DateTime
+import play.api.libs.json.Json
 import play.api.mvc.Result
-import play.api.mvc.Results.{ BadRequest, Conflict, Created }
-import uk.gov.hmrc.auth.core.{ AuthorisationException, Enrolments }
+import play.api.mvc.Results.{Conflict, Created, FailedDependency}
+import uk.gov.hmrc.auth.core.{AuthorisationException, Enrolments}
 import uk.gov.hmrc.http.HeaderCarrier
-import uk.gov.hmrc.securemessage.connectors.EmailConnector
+import uk.gov.hmrc.securemessage.connectors.{ChannelPreferencesConnector, EmailConnector}
 import uk.gov.hmrc.securemessage.controllers.models.generic._
 import uk.gov.hmrc.securemessage.models.EmailRequest
 import uk.gov.hmrc.securemessage.models.core._
+import uk.gov.hmrc.securemessage.models.core.ParticipantType.{Customer => PCustomer, System => PSystem}
 import uk.gov.hmrc.securemessage.repository.ConversationRepository
 
-import scala.concurrent.{ ExecutionContext, Future }
+import scala.concurrent.{ExecutionContext, Future}
 /*
 TODO: refactor so that service has no play dependencies, only core classes.
  */
 
-@SuppressWarnings(Array("org.wartremover.warts.ImplicitParameter"))
-class SecureMessageService @Inject()(repo: ConversationRepository, emailConnector: EmailConnector) {
+@SuppressWarnings(Array("org.wartremover.warts.All"))
+class SecureMessageService @Inject()(repo: ConversationRepository, emailConnector: EmailConnector,
+                                     channelPrefConnector: ChannelPreferencesConnector) {
+
+  type MissingEmailAddress = Boolean
+  private val EmailFound: MissingEmailAddress = true
+  private val EmailNotFound: MissingEmailAddress = !EmailFound
+  private val emailNotFoundErrorMsg = Json.parse("""{"error": 424, "message": "Verified email address could not be found"}""")
 
   def createConversation(conversationRequest: ConversationRequest, client: String, conversationId: String)(
     implicit hc: HeaderCarrier,
-    ec: ExecutionContext): Future[Result] = {
-    val conversation = conversationRequest.asConversation(client, conversationId)
-    repo.insertIfUnique(conversation).map { isUnique =>
+    ec: ExecutionContext): Future[Result] =
+    for {
```

Comment:
you wanna do for comprehension on Either and return an Either as well, propagating the error.

Reply:
another question here is: if one email lookup fails, do we fail all ?